dspace-api/src/main/java/org/dspace/app/sherpa/submit/SHERPASubmitService.java

```diff
@@ -37,12 +38,28 @@ public void setSherpaService(SHERPAService sherpaService) {
         this.sherpaService = sherpaService;
     }
 
-    public SHERPAResponse searchRelatedJournals(Context context, Item item) {
+    public List<SHERPAResponse> searchRelatedJournals(Context context, Item item) {
         Set<String> issns = getISSNs(context, item);
         if (issns == null || issns.size() == 0) {
             return null;
         } else {
-            return sherpaService.searchByJournalISSN(StringUtils.join(issns, ","));
+            // SHERPA v2 API no longer supports "OR'd" ISSN search, perform individual searches instead
+            Iterator<String> issnIterator = issns.iterator();
+            List<SHERPAResponse> responses = new LinkedList<>();
+            while (issnIterator.hasNext()) {
+                String issn = issnIterator.next();
+                SHERPAResponse response = sherpaService.searchByJournalISSN(issn);
+                if (response.isError()) {
+                    // Continue with loop
+                    log.warn("Failed to look up SHERPA ROMeO result for ISSN: " + issn);
```

Comment:
Would it be possible to log the response body as well ? (Not sure if this possible)

Reply:
It's not easily possible here, but what I've done is logged the error message (created upstream when error was set to true in the SHERPAResponse object), and also logged the response body in the SHERPAService and Data Providers if they ever encounter a response code that isn't HttpStatus.SC_OK
See commit: fdc5a3f2

Reply:
Ok